### PR TITLE
Allow RequestLoggingOptions configuration outside of Configure(IApplicationBuilder)

### PR DESCRIPTION
**Pull request by skomis-mm**
It is a common pattern to configure middleware options outside of `Configure(IApplicationBuilder)`. It could be a different module, for example. It also integrates with `Microsoft.Extensions.Options`. This PR will allow such configuration:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.Configure<RequestLoggingOptions>(Configuration.GetSection("Serilog")); // get MessageTemplate from appsettings.json
    services.PostConfigure<RequestLoggingOptions>(options =>
    {
        options.EnrichDiagnosticContext = (diagnosticContext, httpContext) =>
        {
            diagnosticContext.Set("RemoteIpAddress", httpContext.Connection.RemoteIpAddress.MapToIPv4());
        };
    });
}

public void Configure(IApplicationBuilder app)
{
    // ..
    app.UseSerilogRequestLogging();
    app.UseRouting();
    // ...
}
```

### CODE REVIEW

**Comment by skomis-mm** [comment]
@nblumhardt , yes, its pretty complete

File: src/Serilog.AspNetCore/AspNetCore/RequestLoggingOptions.cs [hunk 1]
```diff
@@ -52,13 +62,19 @@ public class RequestLoggingOptions
...
-
         /// <summary>
         /// The logger through which request completion events will be logged. The default is to use the
         /// static <see cref="Log"/> class.
         /// </summary>
         public ILogger Logger { get; set; }
 
-        internal RequestLoggingOptions() { }
+        /// <summary>
+        /// Constructor
```
Comment (sungam3r) [inline src/Serilog.AspNetCore/AspNetCore/RequestLoggingOptions.cs:72]: )

File: src/Serilog.AspNetCore/SerilogApplicationBuilderExtensions.cs [hunk 2]
```diff
@@ -70,12 +62,9 @@ public static class SerilogApplicationBuilderExtensions
             Action<RequestLoggingOptions> configureOptions = null)
```
Comment (sungam3r) [inline src/Serilog.AspNetCore/SerilogApplicationBuilderExtensions.cs:62]: This parameter seems redundant now.

**Comment by mkgn** [comment]
Can RequestLoggingOptions() be enabled/disabled via appSettings.json file under Serilog settings?

**Pull request merged**